controllers/operatorconfig_controller.go

```diff
@@ -55,7 +63,47 @@ type OperatorConfigReconciler struct {
 func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 
-	err := createClusterTasks(ctx, log)
+	// Fetch the OperatorConfig operatorConfig
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, req.NamespacedName, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Add finalizer for this CR
+	if !controllerutil.ContainsFinalizer(operatorConfig, finalizer) {
+		controllerutil.AddFinalizer(operatorConfig, finalizer)
+		err := r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if operatorConfig.DeletionTimestamp != nil {
+		// clean up
+		err := r.CleanUpOpjects(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		controllerutil.RemoveFinalizer(operatorConfig, finalizer)
+		err = r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	err := r.createClusterTasks(ctx, log, operatorConfig)
```

Comment:
this can be refactored with a loop, shared in slack

Reply:
Done